.github/workflows/worker-deploy.yml

```diff
@@ -41,9 +49,117 @@ jobs:
           secrets: |
             SUPABASE_URL
             SUPABASE_KEY
+            ${{ secrets.KERNEL_PUBLIC_KEY && secrets.KERNEL_PUBLIC_KEY != '' && 'KERNEL_PUBLIC_KEY' || '' }}
+
         env:
           SUPABASE_URL: ${{ secrets.SUPABASE_URL }}
           SUPABASE_KEY: ${{ secrets.SUPABASE_KEY }}
+          KERNEL_PUBLIC_KEY: ${{ secrets.KERNEL_PUBLIC_KEY }}
+
+      - name: Update manifest.json worker url
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+
+            const manifestPath = path.resolve("${{ github.workspace }}", './manifest.json');
+            const manifest = JSON.parse(fs.readFileSync(manifestPath, 'utf8'));
+
+            manifest["homepage_url"] = "${{ steps.wrangler_deploy.outputs.deployment-url }}";
```

Comment:
Not necessarily production since it will target the branch it is ran on, so the production one will be the one deploying from the `main` branch. Each branch has its corresponding deployment (we mainly use `development` and `main` at the moment) but any other branch would be also valid. This allows for testing mostly, and switching between dev / prod setup.